ethcore/src/state/mod.rs

```diff
@@ -694,15 +694,26 @@ impl<B: Backend> State<B> {
 		let options = TransactOptions::new(tracer, vm_tracer);
 		let e = self.execute(env_info, engine, t, options, false)?;
 
-		let state_root = if env_info.number < engine.params().eip98_transition || env_info.number < engine.params().validate_receipts_transition {
+		let eip658 = env_info.number >= engine.params().eip658_transition;
+		let no_intermediate_commits = (env_info.number >= engine.params().eip98_transition
+			&& env_info.number >= engine.params().validate_receipts_transition)
+			|| eip658;
```

Comment:
it would be more readable (and efficient) to change the order of those checks and move `eip658 || ... ` to the front